app/controllers/administrative-units/new.js

```diff
@@ -0,0 +1,122 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { dropTask } from 'ember-concurrency';
+import { combineFullAddress } from 'frontend-contact-hub/models/address';
+
+const IDNAMES = {
+  SHAREPOINT: 'SharePoint identificator',
+  KBO: 'KBO nummer',
+};
+
+const CLASSIFICATION = {
+  CENTRAL_WORSHIP_SERVICE: 'f9cac08a-13c1-49da-9bcb-f650b0604054',
+  WORSHIP_SERVICE: '66ec74fd-8cfc-4e16-99c6-350b35012e86',
+};
+
+export default class AdministrativeUnitsNewController extends Controller {
+  @service router;
+  @service store;
+
+  @dropTask
+  *createAdministrativeUnitTask(event) {
+    event.preventDefault();
+
+    let {
+      administrativeUnit,
+      centralWorshipService,
+      worshipService,
+      primarySite,
+      address,
+      contact,
+      identifierSharepoint,
+      identifierKBO,
+      structuredIdentifierSharepoint,
+      structuredIdentifierKBO,
+    } = this.model;
+
+    yield structuredIdentifierSharepoint.save();
+    yield structuredIdentifierKBO.save();
+
+    identifierSharepoint.structuredIdentifier = structuredIdentifierSharepoint;
+    identifierSharepoint.idName = IDNAMES.SHAREPOINT;
+    yield identifierSharepoint.save();
+
+    identifierKBO.structuredIdentifier = structuredIdentifierKBO;
+    identifierKBO.idName = IDNAMES.KBO;
+    yield identifierKBO.save();
+
+    yield contact.save();
+
+    address.fullAddress = combineFullAddress(address);
+    yield address.save();
+
+    primarySite.address = address;
+    primarySite.contacts.pushObjects([contact]);
+    yield primarySite.save();
+
+    administrativeUnit.identifiers.pushObjects([
```

Comment:
What do you mean? Identifiers is a vector.

Reply:
In this code you store the relationships in the temporary `administrativeUnit` record first, and you copy it afterwards. But at this point you can already know which is the correct target record, so this feels like an unnecessary step since you could put it in the correct one directly.